webpack/assets/javascripts/react_app/components/AutoComplete/AutoCompleteActions.js

```diff
@@ -83,7 +96,8 @@ const requestSuccess = ({
 }) => {
   const { error } = data[0] || {};
   if (error) {
-    throw error;
+    // eslint-disable-next-line no-throw-literal
+    throw { message: error, visible: true }; // display this error in the UI.
```

Comment:
Won't it be easier to dispatch an error instead of `throwing an error -> cathing the error -> verify the error type -> dispatch the error`?

Reply:
in this case I am only dispatching the errors which will be visible to the user,
because once the error was passed as a prop, it would be visible..
the errors that I would like to display are only the errors that was return by `success` from the server.
 
another option would be to pass an object similar to the above through the reducer,
and than make this logic in the component - it would keep this mechanism but in my opinion would be a lot messy.

Reply:
Maybe i'm missing something but wouldn't it be the same If you replace the `throw` with:
```js
dispatch(requestFailure({ error, id }));
```

Reply:
gotcha, I thought you were talking about another thing.. :joy:

makes sense, updated :+1: